core/go/internal/txmgr/transaction_submission.go

```diff
@@ -358,18 +358,73 @@ func (tm *txManager) PrepareChainedPrivateTransaction(ctx context.Context, dbTX
 	newTX, err := tm.resolveNewTransaction(ctx, dbTX, tx, submitMode)
 	if err == nil {
 		chained = &components.ChainedPrivateTransaction{
-			OriginalSenderLocator: origSender,
-			OriginalTransaction:   origTxID,
-			OriginalDomain:        origDomain,
+			OriginalSenderLocator: originalSender,
+			OriginalTransaction:   originalTxID,
+			OriginalDomain:        originalDomain,
 			NewTransaction:        newTX,
 		}
-		if origDomainAddress != nil {
-			chained.OriginalContractAddress = origDomainAddress.String()
+		if originalDomainAddress != nil {
+			chained.OriginalContractAddress = originalDomainAddress.String()
 		}
 	}
 	return chained, err
 }
 
+func (tm *txManager) InsertRemoteTransaction(ctx context.Context, dbTX persistence.DBTX, txi *components.ValidatedTransaction, ignoreConflicts bool) (int64, error) {
+	var ptx *persistedTransaction
+	var transactionDeps []*transactionDep
+	// Resolve the finalized fields on the input object for return
+	tx := txi.Transaction
+	tx.Created = pldtypes.TimestampNow()
+	tx.ABIReference = txi.Function.ABIReference
+	tx.Function = txi.Function.Signature
+	// Build the object to insert
+	ptx = &persistedTransaction{
+		ID:             *tx.ID,
+		SubmitMode:     pldapi.SubmitModeRemote.Enum(),
+		Created:        tx.Created,
+		IdempotencyKey: notEmptyOrNull(tx.IdempotencyKey),
+		Type:           tx.Type,
+		ABIReference:   tx.ABIReference,
+		Function:       notEmptyOrNull(txi.Function.Signature),
+		Domain:         notEmptyOrNull(tx.Domain),
+		From:           tx.From,
+		To:             tx.To,
+		Data:           tx.Data,
+	}
+	for _, d := range txi.DependsOn {
+		transactionDeps = append(transactionDeps, &transactionDep{
+			Transaction: *tx.ID,
+			DependsOn:   d,
+		})
+	}
+
+	log.L(ctx).Tracef("insertTransactions to table 'transactions'")
+	insert := dbTX.DB().
+		WithContext(ctx).
+		Table("transactions").
+		Omit("TransactionDeps")
+	if ignoreConflicts {
+		insert = insert.Clauses(clause.OnConflict{DoNothing: true})
+	}
+	txInsertResult := insert.Create(ptx)
+	err := txInsertResult.Error
+	if err == nil && len(transactionDeps) > 0 {
+		log.L(ctx).Debugf("insertTransactions to table 'transaction_deps'")
+		err = dbTX.DB().
+			Table("transaction_deps").
+			Clauses(clause.OnConflict{DoNothing: true}). // for idempotency retry
+			Create(transactionDeps).
+			Error
+	}
+	if err != nil {
+		return -1, err
+	}
+	rowsAffected := txInsertResult.RowsAffected
+
+	return rowsAffected, nil
+}
```

Comment:
This is very significant - a transaction is a decision by users of this node, to do something with their signing key. We've now shared that same DB table with a completely different object - a transaction that is owned by another node, which we somehow want a record of.

If that really is necessary for some reason (feels extremely confusing to me right now), we have to be extremely careful over it.